javascript/apps/taiga-e2e/src/e2e/project/kanban/kanban.cy.ts

```diff
@@ -205,4 +205,57 @@ describe('Kanban', () => {
       cy.get('tg-kanban-status').first().should('not.contain', statusName);
     });
   });
+
+  it('delete status moving stories', () => {
+    cy.getBySel('status-name').first().should('contain', 'New');
+    cy.get('tg-kanban-story').its('length').as('storiesCount');
+    cy.getBySel('status-options').first().click();
+    cy.getBySel('delete-status-btn').should('be.visible');
+    cy.getBySel('delete-status-btn').click();
+    cy.getBySel('submit-delete-status').click();
+
+    cy.getBySel('status-name').first().should('not.contain', 'New');
+    cy.getBySel('status-name').first().should('contain', 'Ready');
+    cy.get<number>('@storiesCount').then(function () {
+      cy.get('tg-kanban-story').should('have.length', this.storiesCount);
+    });
+  });
+
+  it('delete status and stories', () => {
+    cy.getBySel('status-name').first().should('contain', 'New');
+    cy.getBySel('status-options').first().click();
+    cy.getBySel('delete-status-btn').should('be.visible');
+    cy.getBySel('delete-status-btn').click();
+    cy.get('[data-test="delete-stories"]').find('label').click();
+    cy.getBySel('submit-delete-status').click();
+
+    cy.getBySel('status-name').first().should('not.contain', 'New');
+    cy.getBySel('status-name').first().should('contain', 'Ready');
+    cy.get('tg-kanban-story').should('have.length', 0);
+  });
+
+  it('delete all statuses', () => {
+    cy.get('tg-kanban-empty').should('not.exist');
+
+    cy.getBySel('status-options').first().click();
+    cy.getBySel('delete-status-btn').should('be.visible');
+    cy.getBySel('delete-status-btn').click();
+    cy.get('[data-test="delete-stories"]').find('label').click();
+    cy.getBySel('submit-delete-status').click();
+
+    cy.getBySel('status-options').first().click();
+    cy.getBySel('delete-status-btn').should('be.visible');
+    cy.getBySel('delete-status-btn').click();
+
+    cy.getBySel('status-options').first().click();
+    cy.getBySel('delete-status-btn').should('be.visible');
+    cy.getBySel('delete-status-btn').click();
+
+    cy.getBySel('status-options').first().click();
+    cy.getBySel('delete-status-btn').should('be.visible');
+    cy.getBySel('delete-status-btn').click();
+
+    cy.get('tg-kanban-empty').should('exist');
```

Comment:
This assumes that there are ¿five? statuses. It might not be true always.
Also, there is some repeated code

Reply:
This assumes there are 4, same explanation as the other test but, here we are not moving the stories from the first status to the second. We delete he first status + the stories included and then, the rest of them are empty so there isn't a confirmation window and it's deleted straight.
I'll try to refactor the test with a for loop

Reply:
done